python/langsmith/run_trees.py

```diff
@@ -516,6 +520,38 @@ def _get_dicts_safe(self):
             self_dict["outputs"] = self.outputs.copy()
         return self_dict
 
+    def _slice_parent_id(self, parent_id: str, run_dict: dict) -> None:
+        """Slice the parent id from dotted order.
+
+        Additionally check if the current run is a child of the parent. If so, update
+        the parent_run_id to None, and set the trace id to the new root id after
+        parent_id.
+        """
+        if dotted_order := run_dict.get("dotted_order"):
+            segs = _parse_dotted_order(dotted_order)
+            start_idx = None
+            parent_id = str(parent_id)
+            # TODO(angus): potentially use binary search to find the index
```

Comment:
Feels like `segs` will be max like 100 levels deep? Prob not worth thinking about